### PR TITLE
[POC] Firewalld is the new firewall

### PR DESCRIPTION
**Pull request by teclator**
- [Trello Card](https://trello.com/c/oGANpKhQ/1190-5-research-poc-replace-susefirewall2-with-firewalld)
- Moved clients to lib
- Removed support to SuSEFirewall2 in firewall client.
- Started with the new proposal and installation finish clients.

Still [WIP] Need unit tests and open ports add the end.

### CODE REVIEW

File: MAINTAINER [hunk 1]
```diff
@@ -1 +1 @@
-Deprecated file. Use `osc maintainer yast2-firewall` instead.
+Deprecated file. Use `osc maintainer yast2-firewall` instead
```
Comment (imobachgs) [inline MAINTAINER]: np: Do you wanted to remove the point?

File: src/lib/y2firewall/clients/proposal.rb [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        {
+          # Proposal title
+          "rich_text_title" => _("Firewall and SSH"),
+          # Menu entry label
+          "menu_title"      => _("&Firewall and SSH"),
+          "id"              => LINK_FIREWALL_DIALOG
+        }
+      end
+
+      def make_proposal(attrs)
```
Comment (imobachgs) [inline src/lib/y2firewall/clients/proposal.rb]: I do not see the `force_reset`. Is it still needed?

File: src/lib/y2firewall/proposal_settings.rb [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    def enable_firewall!
+      Yast::PackagesProposal.AddResolvables("firewall", :package, ["firewalld"])
+
+      log.info "Enabling Firewall"
+      self.enable_firewall = true
+    end
+
+    def disable_firewall!
+      Yast::PackagesProposal.RemoveResolvables("firewall", :package, ["firewalld"])
```
Comment (imobachgs) [inline src/lib/y2firewall/proposal_settings.rb]: Do you want to uninstall the package? I am not sure if the old code did that.

File: src/lib/y2firewall/widgets/proposal.rb [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      def init
+        self.value = @settings.enable_sshd
+      end
+
+      def label
+        _("Enable SSH Service")
+      end
+
+      def opt
+        %i(notify)
```
Comment (imobachgs) [inline src/lib/y2firewall/widgets/proposal.rb]: I would say that I tend to see [:notify] in YaST (the same applies to %w()).

File: src/lib/y2firewall/clients/proposal.rb (hunk 2 above)
Comment (teclator) [reply src/lib/y2firewall/clients/proposal.rb]: It was not handled in the original client, I could at it reloading the defaults.

File: src/lib/y2firewall/proposal_settings.rb (hunk 3 above)
Comment (teclator) [reply src/lib/y2firewall/proposal_settings.rb]: Same behavior than before:

https://github.com/yast/yast-network/blob/master/src/clients/firewall_stage1_proposal.rb#L404

Comment (imobachgs) [reply src/lib/y2firewall/proposal_settings.rb]: Oh, I see, the stage1 proposal.

**Comment by imobachgs** [comment]
Tests are failing because of missing dependencies. I won't solve the Travis issue but, please, update dependencies if needed.

**Pull request merged**